src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -23,34 +22,28 @@ internal class VsProjectJsonToPackageReferenceMigrator : IVsProjectJsonToPackage
         public VsProjectJsonToPackageReferenceMigrator(Lazy<IVsSolutionManager> solutionManager)
         {
             _solutionManager = solutionManager;
-            if(solutionManager == null)
+            if (solutionManager == null)
             {
                 throw new ArgumentNullException(nameof(solutionManager));
             }
         }
-        public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackageReference(string projectUniqueName)
+        public Task<object> MigrateProjectJsonToPackageReferenceAsync(string projectUniqueName)
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-            if(string.IsNullOrEmpty(projectUniqueName))
+            if (string.IsNullOrEmpty(projectUniqueName))
             {
                 throw new ArgumentNullException(nameof(projectUniqueName));
             }
 
-            if(!File.Exists(projectUniqueName))
+            if (!File.Exists(projectUniqueName))
             {
                 throw new FileNotFoundException(nameof(projectUniqueName));
```

Comment:
The string ctor for this exception takes a *message* rather than a parameter name, per MSDN docs. A message that includes the path to the file that cannot be found can be particularly useful, IMO.